todoapp/app/proguard-rules.pro

```diff
@@ -1,10 +1,18 @@
+-dontoptimize
+
 # Some methods are only called from tests, so make sure the shrinker keeps them.
 -keep class com.example.android.architecture.blueprints.** { *; }
 
--keep class android.support.v4.widget.DrawerLayout { *; }
--keep class android.support.test.espresso.IdlingResource { *; }
--keep class android.support.test.espresso.IdlingRegistry { *; }
+-keep class androidx.drawerlayout.widget.DrawerLayout { *; }
+-keep class androidx.test.espresso.**
+# keep the class and specified members from being removed or renamed
+-keep class androidx.test.espresso.IdlingRegistry { *; }
+-keep class androidx.test.espresso.IdlingResource { *; }
+
 -keep class com.google.common.base.Preconditions { *; }
+
+-keep class androidx.room.RoomDataBase { *; }
```

Comment:
would be great if we could add a comment explaining why each of these needs to be kept. 
can be done in a future PR.